rskj-core/src/main/java/co/rsk/logfilter/BlocksBloomEncoder.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.logfilter;
+
+import org.bouncycastle.util.BigIntegers;
+import org.ethereum.core.Bloom;
+import org.ethereum.util.RLP;
+import org.ethereum.util.RLPList;
+
+import java.math.BigInteger;
+
+/**
+ * Created by ajlopez on 19/02/2020.
+ */
+public class BlocksBloomEncoder {
+    private BlocksBloomEncoder() {
+
+    }
+
+    public static byte[] encode(BlocksBloom blocksBloom) {
+        byte[] rlpFrom = encodeLong(blocksBloom.fromBlock());
+        byte[] rlpTo = encodeLong(blocksBloom.toBlock());
+        byte[] rlpData = RLP.encodeElement(blocksBloom.getBloom().getData());
+
+        return RLP.encodeList(rlpFrom, rlpTo, rlpData);
```

Comment:
should we encode something extra like noBlocks / noConfirmations or a maybe just a version number to ease a future migration if the structure of this changes?

Reply:
Not needed. Only the confirmed blocks are processed in this logic, so each accepted block is processed only once. Check configuration, there is a number of blocks to take as confirmation blocks.

Reply:
Ok, basically if we happen to have to migrate this structure we would have to delete or rename the whole datasource. So we don't need to version this.